1.5/Dockerfile

```diff
@@ -6,12 +6,15 @@ RUN apt-get update && apt-get install -y \
 		--no-install-recommends \
 	&& rm -rf /var/lib/apt/lists/*
 
-ENV GOLANG_GOOS linux
-ENV GOLANG_GOARCH amd64
 ENV GOLANG_VERSION 1.5
+ENV GOLANG_DOWNLOAD_URL https://golang.org/dl/go$GOLANG_VERSION.linux-amd64.tar.gz
+ENV GOLANG_DOWNLOAD_SHA1 5817fa4b2252afdb02e11e8b9dc1d9173ef3bd5a
 
-RUN curl -sSL https://golang.org/dl/go$GOLANG_VERSION.$GOLANG_GOOS-$GOLANG_GOARCH.tar.gz \
-		| tar -v -C /usr/local -xz
+RUN mkdir -p /usr/src/golang \
+	&& curl -sSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz \
+	&& echo "$GOLANG_DOWNLOAD_SHA1  golang.tar.gz" | sha1sum -c - \
+	&& tar -v -C /usr/local -xzf golang.tar.gz \
+	&& rm -r /usr/src/golang
```

Comment:
> Is /usr/src/golang here used for anything?

No, that's why it's being removed. :) It's just to hold the source file to download, then gets removed after.

> (while you're there, mind adding -f to the curl usage and removing -v from tar? :innocent:)

Sure.